### PR TITLE
Adding a method to create an entity naturally

### PR DESCRIPTION
**Pull request by rexbut**
[API](https://github.com/SpongePowered/SpongeAPI/pull/1285) | Common

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java [hunk 1]
```diff
@@ -344,6 +345,15 @@ public void setBiome(int x, int z, BiomeType biome) {
...
     public Optional<Entity> createEntity(EntityType type, Vector3d position) {
+    	return createEntity(type, position, false);
+    }
+    
+    @Override
+    public Optional<Entity> createEntityNaturally(EntityType type, Vector3d position) {
+    	return createEntity(type, position, true);
+    }
+    
+    public Optional<Entity> createEntity(EntityType type, Vector3d position, boolean naturally) {
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: If this isn't overriding any interface, make this private.

File: src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java [hunk 2]
```diff
@@ -400,6 +410,11 @@ public void setBiome(int x, int z, BiomeType biome) {
                 return Optional.empty();
             }
         }*/
+        
+        if (naturally && entity instanceof EntityLiving) {
+            // Adding the default equipment
+        	 ((EntityLiving)entity).onInitialSpawn(world.getDifficultyForLocation(new BlockPos((net.minecraft.entity.Entity) entity)), null);
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: tabs

**Comment by Aaron1011** [comment]
@rexbut: Can you rebase this against `bleeding`?

**Comment by rexbut** [comment]
To fix conflicts, I recreated a new branch : https://github.com/SpongePowered/SpongeCommon/pull/923

**Pull request closed without merging**